tests/zfs-tests/tests/functional/fault/auto_online_001_pos.ksh

```diff
@@ -0,0 +1,151 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+#
+# Copyright (c) 2016 by Intel Corporation. All rights reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/fault/fault.cfg
+
+#
+# DESCRIPTION:
+# Tesing auto-online FMA ZED logic.
+#
+# Method: Creates a pool, exports pool, offlines disk,
+# imports pool with missing disk, then onlines disk.
+# ZED polls for an event change for online disk to be automatically
+# added back to the pool.
+# Creates a raidz1 zpool using persistent disk path names
+# (ie not /dev/sdc). Tests import using pool guid and cache file.
+#
+# Only works for real devices (not loopback)
+
+verify_runnable "both"
+
+disk1=${DISKS%% *}
+if ! ( is_real_device $disk1 || is_mpath_device $disk1 ); then
+	log_unsupported "This test can only be run on real disks."
+fi
```

Comment:
If you move this block to `setup.sh` the whole test group can be cleanly skipped.  Although I think we can probably find a way to run most/all of these tests without real block devices.  The `scsi_debug.ko` module provides a convenient way to create a virtual real scsi device and a mechanism to cause interesting failures.